File: july2020/Day15_reverse_words_in_string.py
class Solution:
  def reverseWords(self, s: str) -> str:
      
    # remove leading/trailing space 
    s = s.strip()
    # remove multiple white spaces in between words
    string = re.sub("\s\s+", " ", s)
    char_arr = string.split(" ")
    ans = ""
    for word in char_arr[::-1]:
        ans += word + " "
    return ans.rstrip()

if __name__ == "__main__":
    obj = Solution()
    print(obj.reverseWords("foo bar"))
        

File: july2020/Day15_reverse_words_in_string.java
package july2020;

class Solution {
  public String reverseWords(String s) {
      String ans = "";
      String after = s.trim().replaceAll(" +", " "); // remove leading/trailing and multiple spaces between words
      
      String[] arr = after.split(" ");
      int n = arr.length -1;
      
      while(n != -1)
      {
          ans = ans + arr[n] + " ";
          n--;
      }
      return ans.trim();       
  }
}


public class Day15_reverse_words_in_string{

  public static void main(String[] args) {
    Solution obj = new Solution();
    System.out.println(obj.reverseWords("  foo bar !"));
  }

}